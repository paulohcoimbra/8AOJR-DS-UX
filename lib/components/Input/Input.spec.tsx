import React from 'react';
import { Input } from ".";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { describe, it, expect} from "vitest";

describe("@component/Input", () => {
  it("should render input with label", () => {
    render(<Input labelText='Username'/>);
    const labelText = screen.getByText("Username");
    expect(labelText).toBeInTheDocument();
  })

  it("should render input with error message", () => {
    render(<Input labelText='Username' errorMessage='Username is required' />);
    const errorMessage = screen.getByRole("alert");
    expect(errorMessage).toHaveTextContent("Username is required");
  })

  it("should match to snapshot", () => {
    const { asFragment } = render(<Input labelText="Username"/>);
    expect(asFragment()).toMatchSnapshot();
  })
})