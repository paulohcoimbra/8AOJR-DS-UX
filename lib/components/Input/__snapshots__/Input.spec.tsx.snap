// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@component/Input > should match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gIIpAQ WalyP"
  >
    <label
      for=""
    >
      <p
        class="sc-glJfXD eqNloQ"
      >
        Username
      </p>
    </label>
    <div
      class="sc-kFkjun LjmJv"
    >
      <input />
    </div>
  </div>
</DocumentFragment>
`;
